hotdoc/extensions/c/utils.py

```diff
@@ -28,14 +20,13 @@ def __init__(self, extension, comment_parser):
 
     def parse_comments(self, filenames):
         for filename in filenames:
-            with open(filename, 'rb') as f:
+            with open(filename, 'r', encoding='utf-8') as f:
```

Comment:
`open()` in text mode with the default `errors="strict"` mode _can_ raise a `ValueError` exception, if the input is _not_ UTF-8 encoded and contains data that can't be decoded.

Just letting them be raised upward to the caller IS an option. Alternatives:
* Pass `errors="ignore"` to `open()` (and potentially lose input data)
* Wrap the whole `with open...` in a `try:` block, and catch any `ValueError`s it throws.

Reply:
I'm not sure what we can really do if there are errors, and I _think_ it's okay to surface those rather than lose input. Perhaps I should do the `try/catch`, log an error, and then let it bubble up? I've done that for now.

Reply:
Sorry for the late review, to be honest I'd rather go back to the original version without the try except block as it doesn't really add anything except indentation :) I'll merge once that's in, I tested the gst docs and meson docs and they built just fine for the record

Reply:
done!